### PR TITLE
Add ability to archive organization_users

### LINKED ISSUE #131: Feature: Archive organization users

### PR DESCRIPTION
**Pull request by JibranKalia**
Closes #131

### CODE REVIEW

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx [hunk 1]
```diff
@@ -54,20 +57,34 @@ class ManageOrgUsers extends React.Component {
...
 
+  archiveOrgUser = (id) => {
+    this.setState({ archivingUser: id });
+
+    organizationUserService
+      .archive(id)
+      .then(() => {
+        toast.success('User has been removed', { hideProgressBar: true, position: 'top-center' });
+        this.setState({ archivingUser: null });
+        this.fetchUsers();
```
Comment (JibranKalia) [inline frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:74]: I am not as comfortable with React as I am with Ruby. It seems a little ugly to re-fetch the list after archiving the user. However, I am struggling to figure out how to do it. This is getting the job done.

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx [hunk 2]
```diff
@@ -54,20 +57,34 @@ class ManageOrgUsers extends React.Component {
...
+      })
+      .catch(({ error }) => {
+        toast.error(error, { hideProgressBar: true, position: 'top-center' });
+        this.setState({ archivingUser: null });
+      });
+  };
+
   createUser = () => {
     this.setState({
-      creatingUser: true
```
Comment (JibranKalia) [inline frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:65]: Ran prettier on this file before modifying it. The diff is a lot easier to read with whitespace off.

File: app/models/organization_user.rb [hunk 3]
```diff
@@ -4,6 +4,8 @@ class OrganizationUser < ApplicationRecord
   belongs_to :organization
   belongs_to :user
 
+  enum status: { invited: "invited", archived: "archived" }
```
Comment (Navaneeth-pk) [inline app/models/organization_user.rb:7]: Shouldn't there be an `active` status? when the user accepted the invitation?

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx (hunk 1 above)
Comment (Navaneeth-pk) [reply frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:74]: You are right. There are a few ways to do this without an additional query. Let's create an issue for this and prioritise later?

File: app/controllers/organization_users_controller.rb [hunk 4]
```diff
@@ -35,4 +35,10 @@ def change_role
     authorize org_user
     org_user.update(role: params[:role])
   end
+
+  def archive
+    org_user = OrganizationUser.find params[:organization_user_id]
+    authorize org_user
```
Comment (subintp) [inline app/controllers/organization_users_controller.rb:41]: This is common to all controller methods. Can you move this `before_action` filter?

File: app/models/organization_user.rb (hunk 3 above)
Comment (subintp) [reply app/models/organization_user.rb:7]: Key and value aresame for enum. Why are defining it as hash

You could define it like this 

` enum status: [ :active, :invited, :archived ]`

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx (hunk 1 above)
Comment (JibranKalia) [reply frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:74]: Absolutely. Will make an issue as soon as this is merged.

File: app/models/organization_user.rb (hunk 3 above)
Comment (JibranKalia) [reply app/models/organization_user.rb:7]: So I might be missing something but when using Rails enums it uses int values. 

https://rails.rubystyle.guide/#enums

File: app/controllers/organization_users_controller.rb (hunk 4 above)
Comment (JibranKalia) [reply app/controllers/organization_users_controller.rb]: Done.

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx [hunk 5]
```diff
@@ -54,20 +57,34 @@ class ManageOrgUsers extends React.Component {
...
       });
   };
 
+  archiveOrgUser = (id) => {
+    this.setState({ archivingUser: id });
+
+    organizationUserService
+      .archive(id)
+      .then(() => {
+        toast.success('User has been removed', { hideProgressBar: true, position: 'top-center' });
```
Comment (Navaneeth-pk) [inline frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:72]: Shouldn't it be "The user has been archived"?

**Pull request merged**